cli/src/commands/log.rs

```diff
@@ -49,8 +49,15 @@ pub(crate) struct LogArgs {
     /// Limit number of revisions to show
     ///
     /// Applied after revisions are filtered and reordered.
-    #[arg(long, short)]
+    #[arg(long, short = 'n')]
     limit: Option<usize>,
+    #[arg(
+        short = 'l',
+        hide = true,
+        conflicts_with = "limit",
+        value_name = "LIMIT"
+    )]
+    deprecated_limit: Option<usize>,
```

Comment:
nit: add comment when this flag will be deleted?
https://github.com/martinvonz/jj/pull/3865/files#diff-f3222d6aa28c803b1faa3f5a9090851c6a9d9d6e59e10f6a1d570172147d9915R53-R57

Reply:
Done, seems to me it makes sense to pick the same version as the referenced pull